tests/orion/schemas/test_schedules.py

```diff
@@ -647,6 +647,52 @@ async def test_rrule_validates_rrule_str(self):
         with pytest.raises(ValidationError, match="(invalid 'FREQ': DAILYBAD)"):
             RRuleSchedule(rrule="FREQ=DAILYBAD")
 
+    async def test_rrule_schedule_handles_complex_rrulesets(self):
+        s = RRuleSchedule(
+            rrule="DTSTART:19970902T090000\n"
+            "RRULE:FREQ=YEARLY;COUNT=2;BYDAY=TU\n"
+            "RRULE:FREQ=YEARLY;COUNT=1;BYDAY=TH\n"
+        )
+        dates_from_1900 = await s.get_dates(5, start=datetime(1900, 1, 1, tz="UTC"))
+        dates_from_2000 = await s.get_dates(5, start=datetime(2000, 1, 1, tz="UTC"))
+        assert len(dates_from_1900) == 3
+        assert len(dates_from_2000) == 0
+
+    @pytest.mark.xfail(reason="we currently cannot roundtrip RRuleSchedule objects")
+    async def test_rrule_schedule_handles_rruleset_roundtrips(self):
+        s1 = RRuleSchedule(
+            rrule="DTSTART:19970902T090000\n"
+            "RRULE:FREQ=YEARLY;COUNT=2;BYDAY=TU\n"
+            "RRULE:FREQ=YEARLY;COUNT=1;BYDAY=TH\n"
+        )
+        s2 = RRuleSchedule.from_rrule(s1.to_rrule())
+        s1_dates = await s1.get_dates(5, start=datetime(1900, 1, 1, tz="UTC"))
+        s2_dates = await s2.get_dates(5, start=datetime(1900, 1, 1, tz="UTC"))
+        assert s1_dates == s2_dates
+
+    async def test_rrule_schedule_rejects_rrulesets_with_many_dtstarts(self):
```

Comment:
Hmm, is this actually the test for many timezones?  Or am I misreading this (since the two dtstarts are actually the same UTC time)?

I was expecting to see two exception tests: one for multiple timezones and one for multiple dtstarts.